<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>
<title>Test of mermaid</title>
</head>
<script src="load-mathjax.js" async></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mermaid/8.3.1/mermaid.min.js" crossorigin="anonymous"></script>
<script>mermaid.initialize({startOnLoad:true});</script>
<script>
let graphStr = `
graph TB;
setup ==> 1 ==> 2; 
    1 --> idmap.do; 1 --> wgs.wrap;
    2 --> 2.1 --> spa.sb; 2.1 --> bgen.sb;
    2 --> 2.2 --> wes.R; 2.2 --> wgs.R; 2.2 --> weswgs.R; 2.2 --> rva.sb; 2.2 --> prune.wrap`;
onload = () => {
  mermaid.render("mermaid", graphStr, document.getElementsByTagName("div")[0]);
}
</script>
<style>
 .mermaid {width:200% !important}
</style>
<body><h2>Programs</h2>
<figure>
<table>
<thead>
<tr><th>Sequence</th><th>Filename</th><th>Description</th></tr>
</thead>
<tbody>
<tr><td>1</td><td>weswgs.sh</td><td>WES/WGS preprocessing</td></tr>
<tr><td>2.1</td><td>spa.sh</td><td>Single-point analysis</td></tr>
<tr><td>2.2</td><td>rva.sh</td><td>Rare-variant analysis</td></tr>
</tbody>
</table>
</figure>
Remarks on variant lists and results submitted centrally are described in INTERVAL.md.</p>

<div class="mermaid" id="mermaidChart0">
graph TB;
setup ==> 1 ==> 2; 
    1 --> idmap.do; 1 --> wgs.wrap;
    2 --> 2.1 --> spa.sb; 2.1 --> bgen.sb;
    2 --> 2.2 --> wes.R; 2.2 --> wgs.R; 2.2 --> weswgs.R; 2.2 --> rva.sb; 2.2 --> prune.wrap;
</div>

A MathJax example from epigenomewide association statistic $z_{EWAS}$ defined through methylation data so that
$$
z_{EWAS} = \frac{w^T_{me}z_T}{\sqrt{w^T_{me}Vw_{me}}}
$$

where $w_{me}$ is the weight associated with methylation. See https://docs.mathjax.org/en/latest/web/configuration.html#local-configuration-file.
</body>
</html>
