<!DOCTYPE html>

<html lang="en">
<head>
<meta charset="utf-8"/>
<meta content="IE=edge" http-equiv="X-UA-Compatible"/>
<meta content="width=device-width, initial-scale=1.0" name="viewport"/>
<link href="../img/favicon.ico" rel="shortcut icon"/>
<title>INTERVAL - SCALLOP-Seq</title>
<link href="../css/bootstrap.min.css" rel="stylesheet"/>
<link href="../css/font-awesome.min.css" rel="stylesheet"/>
<link href="../css/base.css" rel="stylesheet"/>
<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" rel="stylesheet"/>
<script defer="" src="../js/jquery-1.10.2.min.js"></script>
<script defer="" src="../js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/Bash.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/HTML.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/Java.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/JavaScript.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/Perl.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/Python.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/R.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/rust.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/yaml.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
</head>
<body>
<div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
<div class="container">
<a class="navbar-brand" href="..">SCALLOP-Seq</a>
<!-- Expander button -->
<button class="navbar-toggler" data-target="#navbar-collapse" data-toggle="collapse" type="button">
<span class="navbar-toggler-icon"></span>
</button>
<!-- Expanded navigation -->
<div class="navbar-collapse collapse" id="navbar-collapse">
<!-- Main navigation -->
<ul class="nav navbar-nav">
<li class="navitem">
<a class="nav-link" href=".."></a>
</li>
<li class="navitem active">
<a class="nav-link" href="./">INTERVAL</a>
</li>
<li class="navitem">
<a class="nav-link" href="../NOTES/">NOTES</a>
</li>
<li class="dropdown">
<a class="nav-link dropdown-toggle" data-toggle="dropdown" href="#">Tests <b class="caret"></b></a>
<ul class="dropdown-menu">
<li>
<a class="dropdown-item" href="../tests/looping/">Looping</a>
</li>
<li>
<a class="dropdown-item" href="../tests/no_singularity/">No singularity</a>
</li>
<li>
<a class="dropdown-item" href="../tests/cclake/">Singularity</a>
</li>
<li>
<a class="dropdown-item" href="../tests/misc/">Miscellaneous utilities</a>
</li>
</ul>
</li>
</ul>
<ul class="nav navbar-nav ml-auto">
<li class="nav-item">
<a class="nav-link" data-target="#mkdocs_search_modal" data-toggle="modal" href="#">
<i class="fa fa-search"></i> Search
                            </a>
</li>
<li class="nav-item">
<a class="nav-link" href=".." rel="prev">
<i class="fa fa-arrow-left"></i> Previous
                                </a>
</li>
<li class="nav-item">
<a class="nav-link" href="../NOTES/" rel="next">
                                    Next <i class="fa fa-arrow-right"></i>
</a>
</li>
<li class="nav-item">
<a class="nav-link" href="https://github.com/jinghuazhao/SCALLOP-Seq"><i class="fa fa-github"></i> GitHub</a>
</li>
</ul>
</div>
</div>
</div>
<div class="container">
<div class="row">
<div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
<div class="navbar-header">
<button class="navbar-toggler collapsed" data-target="#toc-collapse" data-toggle="collapse" title="Table of Contents" type="button">
<span class="fa fa-angle-down"></span>
</button>
</div>
<div class="navbar-collapse collapse card bg-secondary" id="toc-collapse">
<ul class="nav flex-column">
<li class="nav-item" data-level="1"><a class="nav-link" href="#interval-analysis">INTERVAL analysis</a>
<ul class="nav flex-column">
<li class="nav-item" data-level="2"><a class="nav-link" href="#programs">Programs</a>
<ul class="nav flex-column">
</ul>
</li>
<li class="nav-item" data-level="2"><a class="nav-link" href="#variant-lists-and-results">Variant lists and results</a>
<ul class="nav flex-column">
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div></div>
<div class="col-md-9" role="main">
<h1 id="interval-analysis">INTERVAL analysis</h1>
<h2 id="programs">Programs</h2>
<table>
<thead>
<tr>
<th>Sequence</th>
<th>Filename</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>weswgs.sh</td>
<td>WES/WGS preprocessing</td>
</tr>
<tr>
<td>2.1</td>
<td>spa.sh</td>
<td>Single-point analysis</td>
</tr>
<tr>
<td>2.2</td>
<td>rva.sh</td>
<td>Rare-variant analysis</td>
</tr>
</tbody>
</table>
<p>The hierachy of the scripts is as follows,</p>
<div class="mermaid">graph TB;
setup ==&gt; 1; 1 ==&gt; 2;
    1 --&gt; idmap.do; 1 --&gt; wgs.wrap;
    2 --&gt; 2.1; 2.1 --&gt; spa.sb; 2.1 --&gt; bgen.sb;
    2 --&gt; 2.2; 2.2 --&gt; wes.R; 2.2 --&gt; wgs.R; 2.2 --&gt; weswgs.R; 2.2 --&gt; rva.sb; 2.2 --&gt; prune.wrap;
</div>
<h3 id="comparison-with-alternative-methods">Comparison with alternative methods</h3>
<pre><code class="language-bash">#!/usr/bin/bash

function extract()
{
  export olink=~/rds/projects/olink_proteomics/scallop
  for g in wes wgs
  do
    export IL6A=${olink}/SCALLOP-Seq/work/spa/${g}-cvd3_IL.6RA__P08887.fastGWA.gz
    cat &lt;(gunzip -c ${IL6A} | head -1) &lt;(gunzip -c ${IL6A} | awk '$1==1 &amp;&amp; $10&lt;5e-8' | sort -k1,1n -k3,3n) &gt;&gt; sanger.txt
  done
}

Rscript -e '
  options(width=200)
  library(dplyr)
  sanger &lt;- read.table("sanger.txt",header=TRUE,sep="\t",nrows=11) %&gt;%
            rename(chrpos_grch38=locus..GRCh38.,rsid=rsid..dbSNP_v153,F=AF_original,b=beta,se=standard_error,p=p_value) %&gt;%
            select(chrpos_grch38,REF,ALT,AF,b,se,p)
  wes &lt;- read.table("sanger.txt",header=TRUE,skip=12,nrows=11) %&gt;%
         mutate(chrpos_grch38=paste0("chr",CHR,":",POS)) %&gt;%
         select(chrpos_grch38,BETA,SE,P) %&gt;%
         rename(b_wes=BETA,se_wes=SE,p_wes=P)
  wgs &lt;- read.table("sanger.txt",header=TRUE,skip=24) %&gt;%
         mutate(chrpos_grch38=paste0("chr",CHR,":",POS)) %&gt;%
         rename(b_wgs=BETA,se_wgs=SE,p_wgs=P) %&gt;%
         select(-CHR,-POS,-INFO)
  sanger_wes_wgs &lt;- full_join(sanger,wes) %&gt;% full_join(wgs,by='chrpos_grch38')
  write.table(sanger_wes_wgs,file="sanger-wes-wgs.txt",row.names=FALSE,quote=FALSE,sep="\t")
'
</code></pre>
<h4 id="sangertxt">sanger.txt</h4>
<pre><code>locus (GRCh38)  rsid (dbSNP_v153    REF ALT AF_original beta    standard_error  p_value AC  AN  AF  n_HomRef    n_Het   n_HomAlt    n_pheno phenotype   phenotype_name                  
chr1:154465420  NA  T   C   8.09E-01    5.21E-01    6.22E-02    2.69E-16    1268    1562    8.12E-01    25  244 512 782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154542717  NA  C   T   2.00E-01    -5.44E-01   6.11E-02    3.89E-18    303 1556    1.95E-01    502 249 27  782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154544102  rs3811448   G   A   2.01E-01    -5.43E-01   6.03E-02    1.82E-18    307 1564    1.96E-01    504 249 29  782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154545028  rs3811449   G   A   2.01E-01    -5.44E-01   6.04E-02    1.68E-18    306 1564    1.96E-01    505 248 29  782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154520226  NA  G   A   6.37E-01    4.77E-01    5.01E-02    1.89E-20    995 1546    6.44E-01    98  355 320 782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154552497  rs12144146  A   G   3.11E-01    5.09E-01    5.19E-02    1.86E-21    509 1560    3.26E-01    345 361 74  782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154470484  rs11265621  G   A   6.39E-01    4.87E-01    4.96E-02    1.79E-21    1007    1562    6.45E-01    100 355 326 782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154544001  NA  G   A   3.09E-01    5.06E-01    5.13E-02    1.03E-21    509 1564    3.25E-01    350 355 77  782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154429496  NA  T   C   5.65E-01    8.12E-01    4.11E-02    1.97E-70    886 1564    5.67E-01    148 382 252 782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154435237  rs7521458   T   C   4.08E-01    9.86E-01    3.63E-02    2.12E-114   634 1544    4.11E-01    272 366 134 782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
chr1:154454494  NA  A   C   3.98E-01    1.05E+00    3.57E-02    6.77E-128   628 1550    4.05E-01    270 382 123 782 olinkcvd3_il6ra___p08887    Interleukin-6 receptor subunit alpha (IL-6RA)                   
</code></pre>
<h2 id="variant-lists-and-results">Variant lists and results</h2>
<p>INTERVAL variant lists and results submitted centrally are detailed below..</p>
<h3 id="variant-lists-as-of-11112020">Variant lists as of 11/11/2020</h3>
<p>The following two files were also uploaded.</p>
<ul>
<li>INTERVAL-wes.variantlist.gz</li>
<li>INTERVAL-wgs.variantlist.gz</li>
</ul>
<p>for WES and WGS, respectively.</p>
<h3 id="variant-lists-as-of-10112020">Variant lists as of 10/11/2020</h3>
<ul>
<li>A combination of WES+WGS when there were samples with cvd2/cvd3/inf/neu measurements.</li>
</ul>
<p>To find out which variants were exclusively from WES, we do</p>
<pre><code>gunzip -c INTERVAL-wes+wgs.variantlist.gz | awk '$6&lt;2000' | wc -l
</code></pre>
<p>giving 636,046 out of 116,133,865.</p>
<h3 id="variant-lists-as-of-09112020">Variant lists as of 09/11/2020</h3>
<ul>
<li>
<p>The data contains both WES and WGS genotypes
  -- the sample overlap was only five in the four Olink panels.</p>
</li>
<li>
<p>It is expected that there will be very minor changes with additional work on sample/genotype QC.</p>
</li>
</ul>
<h3 id="multiple-linear-regressions-20-pcs-as-covariates">Multiple linear regressions (+20 PCs as covariates)</h3>
<p>They replaced mixed models for 122 proteins + WES and inf1_IL.17A__Q16552 + WGS.</p>
<pre><code>cvd2_ANG.1__Q15389
cvd2_ADAM.TS13__Q76LX8
cvd2_Protein.BOC__Q9BWV1
cvd2_IL.4RA__P24394
cvd2_STK4__Q13043
cvd2_PAR.1__P25116
cvd2_TF__P13726
cvd2_IL1RL2__Q9HB29
cvd2_PDGF.subunit.B__P01127
cvd2_CXCL1__P09341
cvd2_GIF__P27352
cvd2_PIgR__P01833
cvd2_SOD2__P04179
cvd2_FGF.23__Q9GZV9
cvd2_GH__P01241
cvd2_GLO1__Q04760
cvd2_CD84__Q9UIB8
cvd2_PAPPA__Q13219
cvd2_REN__P00797
cvd2_THBS2__P35442
cvd2_XCL1__P47992
cvd2_SORT1__Q99523
cvd2_CCL17__Q92583
cvd2_MMP.7__P09237
cvd2_ITGB1BP2__Q9UKP3
cvd2_Dkk.1__O94907
cvd2_LPL__P06858
cvd2_THPO__P40225
cvd2_BNP__P16860
cvd2_MMP.12__P39900
cvd2_hOSCAR__Q8IYS5
cvd2_TGM2__P21980
cvd2_CA5A__P35218
cvd2_HSP.27__P04792
cvd2_PARP.1__P09874
cvd2_HAOX1__Q9UJM8
cvd3_TNFRSF14__Q92956
cvd3_ITGB2__P05107
cvd3_IL.17RA__Q96F46
cvd3_OPG__O00300
cvd3_SELP__P16109
cvd3_BLM.hydrolase__Q13867
cvd3_PLC__P98160
cvd3_LTBR__P36941
cvd3_Notch.3__Q9UM47
cvd3_PAI__P05121
cvd3_CCL24__O00175
cvd3_SELE__P16581
cvd3_AZU1__P20160
cvd3_IL.6RA__P08887
cvd3_RETN__Q9HD89
cvd3_FAS__P25445
cvd3_MB__P02144
cvd3_OPN__P10451
cvd3_CTSD__P07339
cvd3_Gal.4__P56470
cvd3_CASP.3__P42574
cvd3_CPB1__P15086
cvd3_CHI3L1__P36222
cvd3_ST2__Q01638
cvd3_CTSZ__Q9UBR2
cvd3_MMP.3__P08254
cvd3_PDGF.subunit.A__P04085
cvd3_PECAM.1__P16284
cvd3_NT.pro.BNP__NA
inf1_MCP.3__P80098
inf1_CD244__Q9BZW8
inf1_IL.7__P13232
inf1_IL.17A__Q16552
inf1_CXCL11__O14625
inf1_AXIN1__O15169
inf1_IL.20RA__Q9UHF4
inf1_CST5__P28325
inf1_CXCL1__P09341
inf1_CCL4__P13236
inf1_CD6__Q8WWJ7
inf1_MCP.4__Q99616
inf1_CCL11__P51671
inf1_TNFSF14__O43557
inf1_FGF.23__Q9GZV9
inf1_LIF.R__P42702
inf1_IL.22.RA1__Q8N6P7
inf1_Beta.NGF__P01138
inf1_IL.13__P35225
inf1_MMP.10__P09238
inf1_TNF__P01375
inf1_CD5__P06127
inf1_CXCL6__P80162
inf1_4E.BP1__Q13541
inf1_IL.20__Q9NYY1
inf1_SIRT2__Q8IXJ6
inf1_CD40__P25942
inf1_IFN.gamma__P01579
inf1_FGF.19__O95750
inf1_LIF__P15018
inf1_NRTN__Q99748
inf1_MCP.2__P80075
inf1_CCL25__O15444
inf1_NT.3__P20783
inf1_IL.5__P05113
inf1_ADA__P00813
neu_GDNF___P39905
neu_EZR___P15311
neu_NCAN___O14594
neu_PRTG___Q2VWP7
neu_PLXNB3___Q9ULL4
neu_LRPAP1___P30533
neu_CLEC1B___Q9P126
neu_ADAM23___O75077
neu_RSPO1___Q2MKA7
neu_LGALS8___O00214
neu_FLRT2___O43155
neu_CLEC10A___Q8IUN9
neu_CTSC___P53634
neu_JAM2___P57087
neu_CTSS___P25774
neu_ASAH2___Q9NR71
neu_ULBP2___Q9BZM5
neu_CD300LF___Q8TDQ1
neu_LAT___O43561
neu_NTRK3___Q16288
neu_TNR___Q92752
</code></pre></div>
</div>
</div>
<footer class="col-md-12">
<hr/>
<p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
</footer>
<script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
<script defer="" src="../js/base.js"></script>
<script defer="" src="../javascripts/config.js"></script>
<script defer="" src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script defer="" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script defer="" src="../search/main.js"></script>
<div aria-hidden="true" aria-labelledby="searchModalLabel" class="modal" id="mkdocs_search_modal" role="dialog" tabindex="-1">
<div class="modal-dialog modal-lg">
<div class="modal-content">
<div class="modal-header">
<h4 class="modal-title" id="searchModalLabel">Search</h4>
<button class="close" data-dismiss="modal" type="button"><span aria-hidden="true">×</span><span class="sr-only">Close</span></button>
</div>
<div class="modal-body">
<p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
<form>
<div class="form-group">
<input class="form-control" id="mkdocs-search-query" placeholder="Search..." title="Type search term here" type="search"/>
</div>
</form>
<div id="mkdocs-search-results"></div>
</div>
<div class="modal-footer">
</div>
</div>
</div>
</div><div aria-hidden="true" aria-labelledby="keyboardModalLabel" class="modal" id="mkdocs_keyboard_modal" role="dialog" tabindex="-1">
<div class="modal-dialog">
<div class="modal-content">
<div class="modal-header">
<h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
<button class="close" data-dismiss="modal" type="button"><span aria-hidden="true">×</span><span class="sr-only">Close</span></button>
</div>
<div class="modal-body">
<table class="table">
<thead>
<tr>
<th style="width: 20%;">Keys</th>
<th>Action</th>
</tr>
</thead>
<tbody>
<tr>
<td class="help shortcut"><kbd>?</kbd></td>
<td>Open this help</td>
</tr>
<tr>
<td class="next shortcut"><kbd>n</kbd></td>
<td>Next page</td>
</tr>
<tr>
<td class="prev shortcut"><kbd>p</kbd></td>
<td>Previous page</td>
</tr>
<tr>
<td class="search shortcut"><kbd>s</kbd></td>
<td>Search</td>
</tr>
</tbody>
</table>
</div>
<div class="modal-footer">
</div>
</div>
</div>
</div>
<script src="https://unpkg.com/mermaid@8.8.0/dist/mermaid.min.js"></script><script>mermaid.initialize({});</script></body>
</html>
